cmake/Packaging.cmake

```
@@ -179,3 +256,41 @@ macro(configure_files srcDir destDir)
   endforeach(templateFile)
 
 endmacro(configure_files)
+
+macro(check_is_rpm)
+  #Check if its RPM Like
+  string(REGEX MATCH suse|fedora|rhel RPMTYPE "${DISTRO_LIKE}")
+  string(REGEX MATCH fedora|suse|rhel RPMNAME "${DISTRO_NAME}")
+  if((NOT ("${RPMTYPE}" STREQUAL "")) OR (NOT ("${RPMNAME}" STREQUAL "")))
+
+    if("${DISTRO_NAME}" STREQUAL "opensuse-tumbleweed")
+      set(DISTRO_NAME "opensuse")
+      set(DISTRO_CODENAME "tumbleweed")
+    elseif(("${DISTRO_NAME}" STREQUAL "almalinux"))
+      string(REGEX MATCH [0-9]+ V "${DISTRO_VERSION_ID}")
+      set(RPM_CN "alma${V}")
+    elseif(("${DISTRO_NAME}" STREQUAL "rocky"))
+      string(REGEX MATCH [0-9]+ V "${DISTRO_VERSION_ID}")
+      set(RPM_CN "rocky${V}")
+    endif()
+
+    #Determain the code name to be used
```

```suggestion
    # Determain the code name to be used.
```

Seems like this kind of logic might be duplicated or similar to elsewhere. Shouldn't it be a macro?